modules/040-node-manager/images/early-oom/src/psi-monitor.go

```diff
@@ -35,16 +40,42 @@ const (
 	sysrqOOMCharacter = "f"
 )
 
+var psiUnavailable = prometheus.NewGauge(prometheus.GaugeOpts{
+	Name: "early_oom_psi_unavailable",
+	Help: "Whether PSI subsystem is unavailable on a given system",
+})
+
+func init() {
+	prometheus.MustRegister(psiUnavailable)
+}
+
 func main() {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, syscall.SIGTERM, syscall.SIGINT)
 
+	server := &http.Server{Addr: "127.0.0.1:8080"}
+	http.Handle("/metrics", promhttp.Handler())
+
+	go func() {
+		if err := server.ListenAndServe(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	psiSupportErr := probePSISupport()
+	if psiSupportErr != nil {
+		log.Println(psiSupportErr)
+		psiUnavailable.Set(1)
+
+		sig := <-c
+		shutdown(sig, server)
```

Comment:
One question. When psiSupportErr is not nil, do we set metric and wait SIGTERM from kubelet due to liveness probe fails ?  Is this expected behavior ?

Reply:
Or we should set metric and do nothing in main cycle ? If our pod is never be ready we will get unwanted alerts about it (KubernetesDaemonSetReplicasUnavailable, KubernetesDaemonSetNotUpToDate).

Reply:
I guess we can remove `sig := <-c` and everything would be fine

Reply:
Or maybe we should check PSISupport before starting the http server?

Reply:
No, all is ok. Liveness probe checked from kube-rbac-proxy so this code works as expected. If we remove 70 line, we go to the cycle which tries to do oom work.